cellxgene_schema_cli/cellxgene_schema/atac_seq.py

```diff
@@ -378,71 +395,136 @@ def get_output_file(fragment_file: str, output_file: Optional[str]) -> str:
 
 
 def index_fragment(
+    organism_ontology_term_id: str,
     fragment_file: str,
+    parquet_file: str,
+    tempdir: str,
+    override_write_algorithm: Optional[str] = None,
     output_file: Optional[str] = None,
 ):
     # sort the fragment by chromosome, start coordinate, and stop coordinate, then compress it with bgzip
     bgzip_output_file = get_output_file(fragment_file, output_file)
     bgzip_output_path = Path(bgzip_output_file)
     bgzip_output_path.unlink(missing_ok=True)
+    bgzip_output_path.touch()
+
+    if override_write_algorithm:
+        write_algorithm = write_algorithm_by_callable[override_write_algorithm]
+    elif not shutil.which("bgzip"):  # check if bgzip cli is installed
+        logger.warning("bgzip is not installed, using slower pysam implementation")
+        write_algorithm = write_algorithm_by_callable["pysam"]
+    else:
+        write_algorithm = write_algorithm_by_callable["cli"]
 
-    prepare_fragment(fragment_file, bgzip_output_file)
+    chromosomes = detect_chromosomes(parquet_file)
+    prepare_fragment(chromosomes, organism_ontology_term_id, parquet_file, bgzip_output_file, tempdir, write_algorithm)
     logger.info(f"Fragment sorted and compressed: {bgzip_output_file}")
+    #
     pysam.tabix_index(bgzip_output_file, preset="bed", force=True)
     tabix_output_file = bgzip_output_file + ".tbi"
     logger.info(f"Index file generated: {tabix_output_file}")
 
+    # Validate line count between original and output files
+    logger.info("Validating line count between original and output files")
+
+    # Count lines in parallel to reduce processing time for large files
+    with ThreadPoolExecutor(max_workers=2) as executor:
+        original_future = executor.submit(count_lines_in_compressed_file, fragment_file)
+        output_future = executor.submit(count_lines_in_compressed_file, bgzip_output_file)
+
+        original_line_count = original_future.result()
+        output_line_count = output_future.result()
+
+    if original_line_count != output_line_count:
```

Comment:
should this check go in the `with ThreadPoolExecutor(max_workers=2) as executor:` block if both `original_line_count` and `output_line_count` are set in there?

Reply:
after thinking about it more it's fine in python but would definitely break in other languages lol, and that's why my instinct was to move it. but i don't think it matters.